fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -216,7 +216,8 @@ def _call_oracle(self, lambda_vec):
 
         if self.subsample != None:
             index_sub = _sample(n=self.subsample, weights=redW, random_state=self.random_state)
-            redX_subsampled = self.constraints.X[index_sub, :]
+            index_sub = redW.index.isin(index_sub)
```

Comment:
This is incorrect, because it replaces any repeated samples by a single sample. We need those repetitions.

Reply:
I also noted that the variable `redX_subsamplpyed` is not used in the following code.